parlai/core/torch_agent.py

```diff
@@ -144,26 +195,39 @@ def vectorize(self, obs, add_start=True, add_end=True, truncate=None):
             obs[label_type + '_choice'] = label
         return obs
 
-    def batchify(self, obs_batch, sort=False, is_valid=lambda obs: 'text_vec' in obs):
-        """Creates a batch of valid observations from an unchecked batch, where
-        a valid observation is one that passes the lambda provided to the function.
-        Assumes each observation has been vectorized by vectorize function.
+    def batchify(self, obs_batch, sort=False,
+                 is_valid=lambda obs: 'text_vec' in obs):
```

Comment:
Do we have any idea of when the default `is_valid` implementation may not be useful? I'm inclined to think having a `lambda` as a default parameter is a terrible idea, and this patch might be a good excuse to remove the behavior.

Reply:
yeah I mean already this won't work for image captioning tasks, since there won't be text. I'm not sure if there's a better default.